sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/samples/Readme.md

```diff
@@ -564,8 +564,25 @@ Console.WriteLine("\n\nQuery for raw humidity events over the past 30 seconds.\n
 QueryAnalyzer humidityEventsQueryAnalyzer = client.Queries.CreateEventsQueryAnalyzer(tsId, TimeSpan.FromSeconds(30));
 await foreach (TimeSeriesPoint point in humidityEventsQueryAnalyzer.GetResultsAsync())
 {
-    double? humidityValue = (double?)point.GetValue("Humidity");
-    Console.WriteLine($"{point.Timestamp} - Humidity: {humidityValue}");
+    TimeSeriesValue humidityValue = point.GetValue("Humidity");
+
+    // Figure out what is the underlying type for the time series value. Since you know your Time Series Insights
+    // environment best, you probably do not need this logic and you can skip to directly casting to the proper
+    // type. This logic demonstrates how you can figure out what type to cast to in the case where you are not
+    // too familiar with the property type
+    Type valueType = humidityValue.Type;
```

Comment:
I don't see a reason to create the `valueType` variable. Also, perhaps a switch statement here would read better.

Reply:
I tried using a switch statement. However, I am unable to switch on a 'Type'. I get an error that I should be switching on constant value. For example,
switch (valueTye)
{
 case (typeof(double?)):
...

gives me an error.

Reply:
I looked online on how i can switch on 'Type', I couldn't find anything useful.

Reply:
fair enough